Sources/FacebookCoreTests/GraphRequestQueryItemBuilderTests.swift

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+//
+// You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
+// copy, modify, and distribute this software in source code or binary form for use
+// in connection with the web services and APIs provided by Facebook.
+//
+// As with any software that integrates with the Facebook platform, your use of
+// this software is subject to the Facebook Developer Principles and Policies
+// [http://developers.facebook.com/policy/]. This copyright notice shall be
+// included in all copies or substantial portions of the software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// swiftlint:disable explicit_type_interface multiline_arguments literal_expression_end_indentation force_unwrapping
+
+@testable import FacebookCore
+import XCTest
+
+class GraphRequestQueryItemBuilderTests: XCTestCase {
+
+  var dictionary = [String: AnyHashable]()
+
+  func testBuildingWithEmptyDictionary() {
+    let items = GraphRequestQueryItemBuilder.build(from: dictionary)
+
+    XCTAssertTrue(items.isEmpty,
+                  "Should not build query items from an empty dictionary")
+  }
+
+  func testBuildingWithStringValues() {
+    dictionary = ["Foo": "Bar"]
+    let expectedItems = [URLQueryItem(name: "Foo", value: "Bar")]
+
+    let items = GraphRequestQueryItemBuilder.build(from: dictionary)
+
+    XCTAssertEqual(items, expectedItems,
+                   "Should build query items from string values")
+  }
+
+  func testBuildingWithNumericValues() {
+    dictionary = [
+      "UInt": UInt(1),
+      "Double": Double(1.0),
+      "Float": Float(1.0)
+    ]
+    let expectedItems = [
+      URLQueryItem(name: "UInt", value: "1"),
+      URLQueryItem(name: "Double", value: "1.0"),
+      URLQueryItem(name: "Float", value: "1.0")
+      ].sorted { $0.name < $1.name }
+
+    let items = GraphRequestQueryItemBuilder.build(from: dictionary)
+      .sorted { $0.name < $1.name }
+
+    XCTAssertEqual(items, expectedItems,
+                   "Should be able to build query items from numeric values")
+  }
+
+  func testBuildingWithDataValues() {
+    dictionary = [
```

Comment:
I'm not clear why we would want to build query items that use the text description of Data, rather than the actual Data

Reply:
Yes. I was hoping someone would call this out. I think we want to ignore Data but I wasn't sure. I was hoping that I would encounter something later in the conversion that would shed some more light on this. Can you think of where we might want to send Data as a query parameter and if so, what form it should take?

Reply:
Maybe it would be better to call out the limited set of types we allow here... Perhaps a new type that those conform to? 

Just spitballing but what do you think of access tokens taking parameters in the form of  `[String: TypeThatCanBeStringifiedToUseInAQueryItem]`? 

Then we could add automatic conformance for acceptable types like `String` and `URL` Obviously with a better name for the concept though. 

The issue is that it's difficult to determine the original intent/all the legit cases from the original method signature of `+ (id)convertRequestValue:(id)value` it's defined [here](https://github.com/facebook/facebook-objc-sdk/blob/36d8ef2fee521d3ec814757991e1c1586d38493e/FBSDKCoreKit/FBSDKCoreKit/Internal/FBSDKInternalUtility.m#L103) for more context.

Reply:
Which is a long way of saying that I also don't like what I've done here and am happy to take suggestions. :)